src/index.ts

```diff
@@ -36,8 +36,59 @@ export type NotarizeWaitOptions = NotarizeResult & NotarizeCredentials;
 export type NotarizeStapleOptions = Pick<NotarizeAppOptions, 'appPath'>;
 export type NotarizeOptions = NotarizeStartOptions;
 
-function authorizationArgs(opts: NotarizeCredentials): string[] {
-  if ('appleId' in opts) {
+function isPasswordCredentials(opts: NotarizeCredentials): opts is NotarizePasswordCredentials {
+  const creds = opts as NotarizePasswordCredentials;
+  return creds.appleId !== undefined || creds.appleIdPassword !== undefined;
+}
+
+function isApiKeyCredentials(opts: NotarizeCredentials): opts is NotarizeApiKeyCredentials {
+  const creds = opts as NotarizeApiKeyCredentials;
+  return creds.appleApiKey !== undefined || creds.appleApiIssuer !== undefined;
+}
+
+export function validateAuthorizationArgs(opts: NotarizeCredentials): NotarizeCredentials {
+  const passwordCreds = opts as NotarizePasswordCredentials;
+  const apiKeyCreds = opts as NotarizeApiKeyCredentials;
+
+  const isPassword = isPasswordCredentials(opts);
+  const isApiKey = isApiKeyCredentials(opts);
```

Comment:
```suggestion
  const isApiKey = isApiKeyCredentials(apiKeyCreds);
```

Reply:
Oh oops, I intended to place the type-casted `const`s in their if's instead. So in this case the `opts` parameter is the union type and `isApiKey` will provide the type hint we can use to access its properties.